### PR TITLE
Improve E2E run times by caching go libs [DO NOT REVIEW]

### PR DESCRIPTION
**Pull request by chatton**
## Description

closes: #5001

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -27,9 +27,6 @@ jobs:
       simd-tag: ${{ steps.get-tag.outputs.simd-tag }}
     steps:
       - uses: actions/checkout@v4
-      - uses: actions/setup-go@v4
```
Comment (chatton) [inline .github/workflows/e2e.yaml:30]: unrelated, but no longer required.

File: .github/workflows/e2e-compatibility-workflow-call.yaml [hunk 2]
```diff
@@ -43,6 +43,7 @@ jobs:
           repository: cosmos/ibc-go
       - uses: actions/setup-go@v4
         with:
+          cache-dependency-path: "**/*.sum"
```
Comment (chatton) [inline .github/workflows/e2e-compatibility-workflow-call.yaml:46]: this doesn't seem to be enough on its own, we can probably look into [this](https://github.com/actions/cache)

Comment (DimitrisJim) [reply .github/workflows/e2e-compatibility-workflow-call.yaml:46]: I've had bad experiences with that but it was more than a year ago.

**Comment by chatton** [comment]
closing this PR as not actively working on, we can use it as a reference in the future.

**Pull request closed without merging**